### PR TITLE
doc(kic) add GWAPI redirect instructions

### PR DESCRIPTION
**Pull request by rainest**
### Description

Add GWAPI instructions to the KIC HTTPS redirect guide. Due to https://github.com/Kong/kubernetes-ingress-controller/issues/4890 this uses our vendor annotations.

Adds a TLS configuration include. This creates a certificate and injects configuration to use it into either an Ingress or Gateway. The certificate hostname is configurable. The Gateway and Ingress names are not and expect the standard `echo` Ingress or `kong` Gateway.

GWAPI does not easily allow adding HTTPS configuration without a TLS Listener, and these Listeners _must_ have a certificate as of GWAPI v0.8. We can't use the default Kong certificates if we don't care about the specific certificates, so we need this include for any guides that touch HTTPS-specific functionality.

### Testing instructions

Preview link: https://deploy-preview-6335--kongdocs.netlify.app/kubernetes-ingress-controller/latest/guides/configuring-https-redirect/

https://deploy-preview-6335--kongdocs.netlify.app/kubernetes-ingress-controller/latest/guides/configuring-https-redirect/#configure-an-https-redirect has some issues where navtabs make the CSS very angry.

https://deploy-preview-6335--kongdocs.netlify.app/kubernetes-ingress-controller/latest/guides/getting-started/#add-tls-configuration no change, but now an include.

### Checklist 

- [x] Review label added 
- [x] PR pointed to correct branch (`main` for immediate publishing, or a release branch: e.g. `release/gateway-3.2`, `release/deck-1.17`)

### CODE REVIEW

**Review by Rajakavitha1 — APPROVED** [review]

**Comment by Rajakavitha1** [comment]
Hi @mheap . I tested validated and formatted the guide. Please review and state your approval/merge so that I can port this content to 3.0

**Comment by Rajakavitha1** [comment]
[ingresshttpsredirect.txt](https://github.com/Kong/docs.konghq.com/files/13185565/ingresshttpsredirect.txt)
[gatewayAPIhttpsredirect.txt](https://github.com/Kong/docs.konghq.com/files/13185567/gatewayAPIhttpsredirect.txt)

File: app/_src/kic-v2/guides/configuring-https-redirect.md [hunk 1]
```diff
@@ -1,127 +1,184 @@
...
+the annotations allow you to configure HTTPS-only HTTPRoutes.
+
+## Test the configuration
+
+With the redirect configuration in place, HTTP requests now receive a
+redirect rather than being proxied upstream:
+1. Send a HTTP request.
     ```bash
-    $ curl $PROXY_IP/test/headers -I
+    curl -ksvo /dev/null http://kong.example/echo --resolve kong.example:80:$PROXY_IP 2>&1 | grep -i http
```
Comment (mheap) [inline app/_src/kic-v2/guides/configuring-https-redirect.md:129]:
```suggestion
    curl -ksvo /dev/null -H 'Host: kong.example' $PROXY_IP/echo 2>&1 | grep -i http
```

File: app/_src/kic-v2/guides/configuring-https-redirect.md [hunk 2]
```diff
@@ -1,127 +1,184 @@
...
+```
+{% endnavtab %}
+{% endnavtabs %}
+{% endcapture %}
+{{ the_code | indent }}
+
+> **Note**: The GatewayAPI _does not_ use an [HTTPRequestRedirectFilter](https://gateway-api.sigs.k8s.io/references/spec/#gateway.networking.k8s.io/v1.HTTPRequestRedirectFilter)
+to configure the redirect. Using the filter to redirect HTTP to HTTPS requires
+a separate HTTPRoute to handle redirected HTTPS traffic, which does not mesh
+well with Kong's single route redirect model.
```
Comment (mheap) [inline app/_src/kic-v2/guides/configuring-https-redirect.md:118]:
```suggestion
> **Note**: The GatewayAPI _does not_ use a [HTTPRequestRedirectFilter](https://gateway-api.sigs.k8s.io/references/spec/#gateway.networking.k8s.io/v1.HTTPRequestRedirectFilter)
to configure the redirect. Using the filter to redirect HTTP to HTTPS requires
a separate HTTPRoute to handle redirected HTTPS traffic, which does not align
well with Kong's single route redirect model.
```

File: app/_includes/md/kic/add-tls-conf.md [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+		"port":443,
+		"protocol":"HTTPS",
+		"tls":{
+				"certificateRefs":[{
+				    "group":"",
+					"kind":"Secret",
+					"name":"{{include.hostname}}"
+				}]
+		}
+    }
```
Comment (mheap) [inline app/_includes/md/kic/add-tls-conf.md:35]:
```suggestion
		"tls":{
                    "certificateRefs":[{
                        "group":"",
                        "kind":"Secret",
                        "name":"{{include.hostname}}"
                    }]
                }
    }
```

**Review by mheap — APPROVED** [review]

**Pull request merged**